cs/ccxt/base/Throttler.cs

```diff
@@ -19,30 +21,41 @@ public Throttler(dict config)
             {"delay", 0.001},
             {"cost", 1.0},
             {"tokens", 0},
-            {"maxCapacity", 2000},
+            {"maxLimiterRequests", 2000},
             {"capacity", 1.0},
+            {"algorithm", "leakyBucket"},
+            {"rateLimit", 0.0},
+            {"windowSize", 60000.0},
+            // maxWeight should be set in config parameter for rolling window algorithm
         };
         this.config = extend(this.config, config);
 
     }
 
-    private async Task loop()
+    private async Task leakyBucketLoop()
     {
         var lastTimestamp = milliseconds();
         while (this.running)
         {
             // do we need this check here?
-            if (this.queue.Count == 0)
+            lock (queueLock)
+            {
+                if (this.queue.Count == 0)
+                {
+                    this.running = false;
+                    continue;
+                }
+            }
+            (Task, double) first;
+            lock (queueLock)
             {
-                this.running = false;
-                continue;
+                first = this.queue.Peek();
             }
-            var first = this.queue.Peek();
             var task = first.Item1;
             var cost = first.Item2;
             var tokensAsString = Convert.ToString(this.config["tokens"], CultureInfo.InvariantCulture);
             var floatTokens = double.Parse(tokensAsString, CultureInfo.InvariantCulture);
-            if (floatTokens >= 0)
+            if (floatTokens >= cost)
```

Comment:
@carlosmiei I have added this change, I think this should be correct to continue the permit for request. If you think it should be `>= 0` i'll revert